### DISCUSSION TITLE
prefix in objstoreConfig not working

### DISCUSSION THREAD

**Discussion by hesanjea** [thanos-io/thanos#5442] [Questions & Answers] [answered]
Hi, 
sorry if this is a silly question but I am trying to deploy thanos using the bintnami chart and when trying to configure a prefix as follows:
objstoreConfig:
    type: S3
    config:
      bucket: "bucket-namet"
      endpoint: "s3.eu-west-1.amazonaws.com"
      region: ""
      aws_sdk_auth: false
      access_key: ""
      insecure: false
      signature_version2: false
      secret_key: ""
      put_user_metadata: {}
      http_config:
        idle_conn_timeout: 1m30s
        response_header_timeout: 2m
        insecure_skip_verify: false
        tls_handshake_timeout: 10s
        expect_continue_timeout: 1s
        max_idle_conns: 100
        max_idle_conns_per_host: 100
        max_conns_per_host: 0
        tls_config:
          ca_file: ""
          cert_file: ""
          key_file: ""
          server_name: ""
          insecure_skip_verify: true
      trace:
        enable: false
      list_objects_version: ""
      part_size: 67108864
      sse_config:
        type: ""
        kms_key_id: ""
        kms_encryption_context: {}
        encryption_key: ""
      sts_endpoint: ""
    prefix: "thanos_metrics" 

I am getting this error:

level=error ts=2022-06-24T08:14:54.625680723Z caller=main.go:132 err="yaml: unmarshal errors:\n  line 36: field prefix not found in type client.BucketConfig\nparsing config YAML file\ngithub.com/thanos-io/thanos/pkg/objstore/client.NewBucket\n\t/app/pkg/objstore/client/factory.go:52\nmain.runReceive\n\t/app/cmd/thanos/receive.go:161\nmain.registerReceive.func1\n\t/app/cmd/thanos/receive.go:86\nmain.main\n\t/app/cmd/thanos/main.go:130\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:255\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1581\npreparing receive command failed\nmain.main\n\t/app/cmd/thanos/main.go:132\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:255\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1581"

what am i doing wrong?
Thank you for ur support :)

**Comment by crockk** [comment]
I can't tell what your indentation is but you probably have something like:
```
type: S3
config:
  bucket: "mybucket"
  . . . 
  prefix: "thanos_metrics"
```

Prefix is at the root level in the config hierarchy, so it should be:
```
type: S3
config:
  bucket: "mybucket"
  . . . 
prefix: "thanos_metrics"
```

Try that.

**Reply by hesanjea** [reply]
Thank you @crockk , I'm using the indentation as explained in Thanos doc ("prefix" at the same level than "type" and "config").
https://thanos.io/tip/thanos/storage.md/#s3

**Comment by goodosoft** [comment, marked as answer]
upgrade thanos to v0.28.0 will fix it.

**Reply by hesanjea** [reply]
Thank you @goodosoft  !!!!
